### ISSUE TITLE
Repo manager `check` for variantsets misleading

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#1059]
In a previous fix the "Inconsistent metadata" between VCFs was ignored, however, this is still part of `repo check`. We should either make these checks meaningful or remove them entirely.

**Comment by dcolligan** [comment]
Could you elaborate?  What do you consider a meaningful check?

**Comment by david4096** [comment]
I posted an issue that the server wasn't starting after adding 1kg variant set (https://github.com/ga4gh/server/issues/946). It seems to run fine when I've added it, but `repo check` claims there is something wrong. What feature are we supporting by having repo check fail?

**Comment by kozbo** [comment]
Another way to put it is... Is the ga4gh_repo check failure a false negative? The API seems to function without error (perhaps due to the fix in #946) so we are wondering if the same fix needs got be applied to the logic of the check function (or some similar logic). The alternative is that the error message from the check function is a **real** problem that needs to be fixed.  

I am pasting the error output here for reference.
```
(ga4gh-server-env)azureuser@kobo-vm:/srv/ga4gh$ ga4gh_repo check /srv/ga4gh/ga4gh-example-data
Warning: The index file is older than the data file: /srv/ga4gh/ga4gh-example-data/datasets/1kg-p3-subset/variants/vcf/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz.tbi
Warning: The index file is older than the data file: /srv/ga4gh/ga4gh-example-data/datasets/1kg-p3-subset/variants/vcf/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz.tbi
Warning: The index file is older than the data file: /srv/ga4gh/ga4gh-example-data/datasets/1kg-p3-subset/reads/HG00096.mapped.ILLUMINA.bwa.GBR.low_coverage.20120522.bam.bai
Warning: The index file is older than the data file: /srv/ga4gh/ga4gh-example-data/datasets/1kg-p3-subset/variants/vcf/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz.tbi
Warning: The index file is older than the data file: /srv/ga4gh/ga4gh-example-data/datasets/1kg-p3-subset/variants/vcf/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz.tbi
Traceback (most recent call last):
  File "/srv/ga4gh/ga4gh-server-env/bin/ga4gh_repo", line 9, in <module>
    load_entry_point('ga4gh==0.2.0a2.dev170+ngcce89bc', 'console_scripts', 'ga4gh_repo')()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/cli.py", line 1672, in repo_main
    runner.run()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/cli.py", line 1385, in run
    self.repoManager.check(self.doConsistencyCheck)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/repo_manager.py", line 293, in check
    dataRepo.checkConsistency()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/datarepo.py", line 214, in checkConsistency
    variantSet.checkConsistency()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/datamodel/variants.py", line 580, in checkConsistency
    self._checkMetadata(varFile)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/datamodel/variants.py", line 556, in _checkMetadata
    variantFile.filename)
ga4gh.exceptions.InconsistentMetaDataException: Metadata of /srv/ga4gh/ga4gh-example-data/datasets/1kg-p3-subset/variants/vcf/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz is not consistent with other files in this VariantSet, and inconsistent metadata is not permitted
```

**Comment by dcolligan** [comment]
Yeah so the `check` function (among other things) runs the `checkConsistency` methods which run any number of tests to determine if the data that you want to add is "consistent."  We should definitely evaluate if the checks are in there are still necessary, and if we should add ones that are not there... @jeromekelleher ?

**Comment by jeromekelleher** [comment]
I think the checkConsistency method is intended to make sure that the VCFs we're putting into a VariantSet really are all from the same VariantSet by checking the headers. We should definitely do this at ``add`` time in repo manager. If the VCFs really are from the same VariantSet and we're getting an error then there is a problem with the checking logic and we should file a specific bug against that functionality.

**Comment by david4096** [comment]
The original bug post was about the inconsistent metadata check was failing for the 1kg dataset. https://github.com/ga4gh/server/issues/946

I'm happy to close this and reopen as another issue, as @jeromekelleher states we need to know whether the inconsistent metadata check is meaningful (or not).

**Comment by jeromekelleher** [comment]
Closed by #1166.
